projects/eslint-config/plugins/liferay/lib/rules/sort-imports.js

```diff
@@ -31,21 +32,34 @@ function compare(aKey, bKey) {
 	//
 
 	return (
+		cmp(aPrefix, bPrefix) ||
 		cmp(ranking(aName), ranking(bName)) ||
 		cmp(aName, bName) ||
 		cmp(aTieBreaker, bTieBreaker)
 	);
 }
 
 /**
- * Ideally we'd just sort by source, but in case there is a pathological case
- * where we import from the same module twice, we need to include information
- * about the import specifiers in the sort key.
+ * Ideally we'd just sort by source, but we need to handle two "edge" cases:
+ *
+ * - When we have type-only imports, those always come last.
+ * - When we import from the same module twice (not always an error; can
+ *   legitimately happen if we have both a type-only and a value-only import
+ *   from the same module).
+ *
+ * So, the sort key must include information about both the import kind
+ * (this serves as a prefix in the sort key) and any import specifiers
+ * that may be present (these serve to construct a trailing "tiebreaker"
+ * suffix in the sort key).
  */
 function getSortKey(node) {
 	const source = getSource(node);
 	let tieBreaker;
 
+	// Type-only imports always go last.
+
+	const prefix = isTypeImport(node) ? 'type-import' : 'normal-import';
```

Comment:
Strictly speaking, we don't need to bake this indicator of "type-import-ness" into the key itself (because we are handling these specially further down anyway). I think I could delete this and all the tests would still pass. But, it doesn't really hurt too much for the ordering info to be in the key anyway, so 🤷